### PR TITLE
Test user is not displayed on autocompletion when less than the default minimum characters are typed

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance test to assert autocomplete list is not present when not enough characters are typed

## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 1]
```diff
@@ -467,3 +474,17 @@ Then('the file {string} should be in {string} state on the webUI after a page re
 Then('the autocomplete list should not be displayed on the webUI', function () {
   return client.page.FilesPageElement.sharingDialog().assertAutocompleteListIsNotVisible()
 })
+
+Then('only {string} should be listed in the autocomplete list on the webUI', function (username) {
+  return client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+    .then(itemsList => {
+      console.log(itemsList)
```
Comment (dpakach) [inline tests/acceptance/stepDefinitions/sharingContext.js]: I think Debug console.log made it to the commit.

File: tests/acceptance/features/webUISharingInternalUsers/shareAutocompletion.feature [hunk 2]
```diff
@@ -55,11 +55,26 @@ Feature: Autocompletion of share-with names
...
+    Then the autocomplete list should not be displayed on the webUI
+
+    @skip @yetToImplement
+  Scenario: autocomplete short user/display names when completely typed
+    Given the administrator has set the minimum characters for sharing autocomplete to "2"
+    And user "regularuser" has logged in using the webUI
+    And the user has browsed to the files page
     And these users have been created but not initialized:
       | username | password | displayname | email        |
       | use      | %alt1%   | Use         | uz@oc.com.np |
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingInternalUsers/shareAutocompletion.feature]: What's going on here?

**Comment by jasson99** [comment]
cc : please review @dpakach @skshetry @HariBhandari07

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Files</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6040/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6044/

<p>










</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6071/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6085/

<p>





</p></details>

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 3]
```diff
@@ -283,6 +290,88 @@ Then('all users and groups that contain the string {string} in their name should
...
+          assert.ok(
+            itemsList.includes(groupString),
+            `could not find '${groupString}' in autocomplete share list`
+          )
+        }
+      })
+    })
+})
+
+Then('all users and groups that contain the string {string} in their name should be listed in the autocomplete list on the webUI except group {string}', function (pattern, alreadySharedGroup) {
```
Comment (HariBhandari07) [inline tests/acceptance/stepDefinitions/sharingContext.js]: This function looks exactly the same like the above function, find a way to minimize the duplicate codes

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RestrictSharing</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6149/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6149/

<p>






































</p></details>

**Comment by jasson99** [comment]
@skshetry @HariBhandari07 @dpakach

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6149/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6149/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6150/

<p>




























</p></details>

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -494,3 +559,16 @@ Then('the file {string} should be in {string} state on the webUI after a page re
 Then('the autocomplete list should not be displayed on the webUI', function () {
   return client.page.FilesPageElement.sharingDialog().assertAutocompleteListIsNotVisible()
 })
+
+Then('only {string} should be listed in the autocomplete list on the webUI', function (username) {
+  return client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+    .then(itemsList => {
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Why not `assert` that there is only 1 item and the first item is the expected item? I'd think `only` as having only one item.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6244/

<p>











































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6244/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6248/

<p>













</p></details>

**Comment by jasson99** [comment]
@skshetry @dpakach @HariBhandari07

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 5]
```diff
@@ -160,14 +162,66 @@ const checkIsListedInAutoComplete = function (type, displayName) {
...
+      return displayName !== currentUserDisplayName && displayName !== alreadySharedUserOrGroup
+    }
+  )
+  const groupPostfix = client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+  const userPostfix = client.page.FilesPageElement.sharingDialog().getUserSharePostfix()
+  const usersNotFound = _.difference(
+    usersMatchingPattern.map(name => name + userPostfix),
+    itemsList
+  )
+  assert.strictEqual(usersNotFound.length, 0, `could not find ${usersNotFound} in the itemsList`)
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Please split these into two functions:
1. that asserts the given users are listed in the auto-completion list.
2. that asserts the given groups are listed in the auto-completion list.

And, also adjust the other function (the one *without* `exclude`) and this function accordingly.

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 6]
```diff
@@ -160,14 +162,78 @@ const checkIsListedInAutoComplete = function (type, displayName) {
...
+  const itemsList = await client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+  const groupPostfix = client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+  const groupsNotFound = _.difference(
+    groupMatchingPattern.map(name => name + groupPostfix),
+    itemsList
+  )
+  assert.strictEqual(groupsNotFound.length, 0, `could not find ${groupsNotFound} in the itemsList`)
+}
+
+const getUsersWithThePattern = async function (pattern, alreadySharedUserOrGroup, userOrGroup) {
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Looks great. You just need to fix the name of this function.

```suggestion
const assertUsersGroupsWithPatternInAutocompleteListExcluding = async function (pattern, sharedUserOrGroup, userOrGroup) {
```

Comment (skshetry) [reply tests/acceptance/stepDefinitions/sharingContext.js]: If you have shorter name than this, please comment. :grin:

Comment (jasson99) [reply tests/acceptance/stepDefinitions/sharingContext.js]: Thankyou.I fixed that.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6311/

<p>























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6311/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6316/

<p>

































</p></details>

**Pull request merged**